### ISSUE TITLE
Broad Training: Design training loop

### ISSUE THREAD

**Issue by bclavie** [AnswerDotAI/bert24#19]
Broad issue that will eventually be split up. We need to experiment and implement what makes the most sense in terms of:

- LR Schedule: Warmup-Stable-Decay or Reciprocal SQRT + Cosine to facilitate future continuous pre-training?
- Context length ramp-up, to support our target context sizes (4k target for now)
- Batch size ramp-up and figuring out what the best tradeoff is

**Comment by ohallstrom** [comment]
WSD has been added in the following PR: #31. I can have a go at the ramp-ups as well
